src/iuse.cpp

```diff
@@ -2012,41 +2012,79 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
 
     // TODO: Wrap all the mods into structs, maybe even json-able
     // All possible mods here
-    std::array<std::string, 4> clothing_mods{
+    const std::array<const std::string, 4> clothing_mods{
         { "wooled", "furred", "leather_padded", "kevlar_padded" }
     };
 
     // Materials those mods use
-    std::array<std::string, 4> mod_materials{
+	const std::array<const std::string, 4> mod_materials{
         { "felt_patch", "fur", "leather", "kevlar_plate" }
     };
 
+    // TODO: Read data from \data\json\items\generic\filament.json
+    // Filament that can be used
+	const std::array<const std::string, 4> mod_filament{
+		{ "thread", "sinew", "plant_fibre", "yarn" }
+	};
```

Comment:
I've already added `filament` requirement earlier  when I changed recipes in #20270. I will try to implement its usage in sew_advanced just like you described. Thanks!